src/main/java/org/opensearch/ad/transport/RCFPollingTransportAction.java

```diff
@@ -19,59 +19,63 @@
 import org.apache.logging.log4j.Logger;
 import org.opensearch.action.ActionListener;
 import org.opensearch.action.support.ActionFilters;
-import org.opensearch.action.support.HandledTransportAction;
-import org.opensearch.ad.cluster.HashRing;
+import org.opensearch.action.support.TransportAction;
 import org.opensearch.ad.common.exception.AnomalyDetectionException;
 import org.opensearch.ad.ml.ModelManager;
 import org.opensearch.ad.ml.SingleStreamModelIdMapper;
 import org.opensearch.ad.settings.AnomalyDetectorSettings;
 import org.opensearch.cluster.node.DiscoveryNode;
-import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.inject.Inject;
 import org.opensearch.common.io.stream.StreamInput;
-import org.opensearch.common.settings.Settings;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.sdk.SDKClusterService;
 import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskManager;
 import org.opensearch.threadpool.ThreadPool;
 import org.opensearch.transport.TransportException;
 import org.opensearch.transport.TransportRequestOptions;
 import org.opensearch.transport.TransportResponseHandler;
 import org.opensearch.transport.TransportService;
 
+import com.google.inject.Inject;
+
 /**
  * Transport action to get total rcf updates from hosted models or checkpoint
  *
  */
-public class RCFPollingTransportAction extends HandledTransportAction<RCFPollingRequest, RCFPollingResponse> {
+public class RCFPollingTransportAction extends TransportAction<RCFPollingRequest, RCFPollingResponse> {
 
     private static final Logger LOG = LogManager.getLogger(RCFPollingTransportAction.class);
     static final String NO_NODE_FOUND_MSG = "Cannot find model hosting node";
     static final String FAIL_TO_GET_RCF_UPDATE_MSG = "Cannot find hosted model or related checkpoint";
 
     private final TransportService transportService;
     private final ModelManager modelManager;
-    private final HashRing hashRing;
+    // private final HashRing hashRing;
     private final TransportRequestOptions option;
-    private final ClusterService clusterService;
+    private final SDKClusterService sdkClusterService;
+    // private final DiscoveryNode discoveryNode;
 
     @Inject
     public RCFPollingTransportAction(
         ActionFilters actionFilters,
-        TransportService transportService,
-        Settings settings,
+        // Settings settings,
         ModelManager modelManager,
-        HashRing hashRing,
-        ClusterService clusterService
```

Comment:
same here with the cluster service

Reply:
Replied the reason in the comment before.